### PR TITLE
cli: limit flag for debug range-data / debug keys

### PR DESCRIPTION
**Pull request by andreimatei**
There was a teasing maxResults field already, completely unused.

Release note: None

### CODE REVIEW

**Review by irfansharif — APPROVED** [review]
:lgtm: 

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49290#-:-M7iypHQ7Pb0NwOE3xWt:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei and @tbg)

___
*[pkg/cli/debug.go, line 304 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49290#-M7ixsbc6qzDWw2_alzD:-M7ixsbc6qzDWw2_alzE:b-bl9nx8) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e4c5c8345e02a89e349a88f331e967391f62909/pkg/cli/debug.go#L304)):*
> ```Go
> 	defer iter.Close()
> 	for i := 0; ; iter.Next() {
> 		i++
> ```

[nit] Move the `i` declaration outside the for loop block, s/i/results, and move `results++` closer to the limit check (similar to what we now have for `runDebugKeys`).

File: pkg/cli/context.go [hunk 1]
```diff
@@ -105,7 +105,6 @@ func initCLIDefaults() {
 	debugCtx.replicated = false
 	debugCtx.inputFile = ""
 	debugCtx.printSystemConfig = false
-	debugCtx.maxResults = 1000
```
Comment (knz) [inline pkg/cli/context.go:108]: Keep this line; it's needed. You can change the value to zero.

File: pkg/cli/flags.go [hunk 2]
```diff
@@ -697,12 +697,14 @@ func init() {
 		f := debugKeysCmd.Flags()
 		VarFlag(f, (*mvccKey)(&debugCtx.startKey), cliflags.From)
 		VarFlag(f, (*mvccKey)(&debugCtx.endKey), cliflags.To)
+		IntFlag(f, &debugCtx.maxResults, cliflags.Limit, 0)
```
Comment (knz) [inline pkg/cli/flags.go:700]: last arg is `debugCtx.maxResults`, not `0`

File: pkg/cli/flags.go [hunk 3]
```diff
@@ -697,12 +697,14 @@ func init() {
...
 		BoolFlag(f, &debugCtx.values, cliflags.Values, debugCtx.values)
 		BoolFlag(f, &debugCtx.sizes, cliflags.Sizes, debugCtx.sizes)
 	}
 	{
 		f := debugRangeDataCmd.Flags()
 		BoolFlag(f, &debugCtx.replicated, cliflags.Replicated, debugCtx.replicated)
+		IntFlag(f, &debugCtx.maxResults, cliflags.Limit, 0)
```
Comment (knz) [inline pkg/cli/flags.go:707]: ditto

**Review by knz — CHANGES_REQUESTED** [review]

File: pkg/cli/context.go (hunk 1 above)
Comment (andreimatei) [reply pkg/cli/context.go:108]: done

**Review by knz — COMMENTED** [review]
> Reviewed 4 of 4 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49290)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @andreimatei)

___
*[pkg/cli/flags.go, line 700 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49290#-M7iq-LLNfV9QNLfAhtz-r2-700:-M8e1hfTEUHIW9MnQpP2:b-r275my) ([raw file](https://github.com/cockroachdb/cockroach/blob/0dc131ec33e24a88b689bc96ba5839e9eeaf950f/pkg/cli/flags.go#L700)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

last arg is `debugCtx.maxResults`, not `0`
</blockquote></details>

that still needs to change (we don't want to reproduce defaults in multiple places)
___
*[pkg/cli/flags.go, line 707 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49290#-M7iq-LLNfV9QNLfAhtz-r2-707:-M8e1j_58V3a5dM38MWp:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/0dc131ec33e24a88b689bc96ba5839e9eeaf950f/pkg/cli/flags.go#L707)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

ditto

**Review by andreimatei — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49290)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @irfansharif and @knz)

___
*[pkg/cli/debug.go, line 304 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49290#-M7ixsbc6qzDWw2_alzD:-M8W2em40VVng-8RFoSf:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e4c5c8345e02a89e349a88f331e967391f62909/pkg/cli/debug.go#L304)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

[nit] Move the `i` declaration outside the for loop block, s/i/results, and move `results++` closer to the limit check (similar to what we now have for `runDebugKeys`).
</blockquote></details>

done
___
*[pkg/cli/flags.go, line 700 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49290#-M7iq-LLNfV9QNLfAhtz-r2-700:-M8kT3vCC5JbCCI5kZZ9:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0dc131ec33e24a88b689bc96ba5839e9eeaf950f/pkg/cli/flags.go#L700)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

that still needs to change (we don't want to reproduce defaults in multiple places)
</blockquote></details>

Done.
___
*[pkg/cli/flags.go, line 707 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/49290#-M7iq-LLNfV9QNLfAhtz-r2-707:-M8kT4DW5JRuljpbMA-x:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/0dc131ec33e24a88b689bc96ba5839e9eeaf950f/pkg/cli/flags.go#L707)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

**Review by knz — APPROVED** [review]
thx

**Review by andreimatei — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49290#-:-M8l0CV49N0dtwzrs2fF:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @irfansharif and @knz)

**Pull request merged**